src/utils/word-api.ts

```diff
@@ -101,31 +95,52 @@ class WordApi {
     return `JABREF-CITATION-${JSON.stringify(citation)}`;
   };
 
-  static async getTotalNumberOfCitations(): Promise<number | null> {
-    const jabRefCitations =
-      (await WordApi.getJabRefCitations()) as Array<Word.ContentControl>;
-    return jabRefCitations.length;
+  static async getTotalNumberOfCitations(): Promise<number | void> {
+    return Word.run(async (context: Word.RequestContext) => {
+      const jabRefCitations = await WordApi.getJabRefCitations(context);
+      return jabRefCitations.length;
+    }).catch((error) => {
+      console.log(`Error: ${JSON.stringify(error)}`);
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
+    });
   }
 
   static async getCitationIdToPos(): Promise<Record<string, number> | void> {
-    const jabRefCitations =
-      (await WordApi.getJabRefCitations()) as Array<Word.ContentControl>;
-    const citationIdToPos: Record<string, number> = {};
-    let pos = 0;
-    jabRefCitations.forEach((citation) => {
-      const { tag } = citation;
-      citationIdToPos[tag.substring(16)] = pos;
-      pos += 1;
+    return Word.run(async (context: Word.RequestContext) => {
+      const jabRefCitations = await WordApi.getJabRefCitations(context);
+      const citationIdToPos: Record<string, number> = {};
+      jabRefCitations.forEach((citation, index) => {
+        const { tag } = citation;
+        citationIdToPos[tag.substring(16)] = index;
+      });
+      return citationIdToPos;
+    }).catch((error) => {
+      console.log(`Error: ${JSON.stringify(error)}`);
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
     });
-    return citationIdToPos;
   }
 
-  static async getCitationByIndex(): Promise<Array<StatefulCitation> | null> {
-    const jabRefCitations =
-      (await WordApi.getJabRefCitations()) as Array<Word.ContentControl>;
-    return jabRefCitations.map(
-      (citation) => JSON.parse(citation.tag.substring(16)) as StatefulCitation
-    );
+  static async getCitationByIndex(): Promise<Array<StatefulCitation> | void> {
+    return Word.run(
+      async (
```

Comment:
For lambdas you usually don't need to annotate the types since they can be inferred from the context:
`Word.run( async (context) => { ... })`